lib/dpl/provider/pages.rb

```diff
@@ -49,6 +50,7 @@ def initialize(context, options)
         @gh_ref = "#{@gh_url}/#{slug}.git"
         @git_push_opts = @keep_history ? '' : ' --force'
         @git_commit_opts = (@allow_empty_commit and @keep_history) ? ' --allow-empty' : ''
+        @commit_message = Shellwords.escape(options[:commit_message]) || "Deploy #{@project_name} to #{@gh_ref}:#{@target_branch}"
```

Comment:
Hey @sarahdayan, afraid this will resolve _commit_message_ to be an empty string in the case of an unspecified custom commit message since `Shellwords.escape(options[:commit_message])`  will evaluate to a _"''"_ resulting true.

```console
ubuntu@0fbb6a31ea12:~/workspace/assessment$ dpl --skip_cleanup=true --provider=pages --github-token="$GITHUB_TOKEN"
Installing deploy dependencies

!!! GitHub Pages support is experimental !!!

Preparing deploy
Logged in as @ashenm (Ashen Gunaratne)
Deploying application
cd /tmp/d20190730-915-x8jbz9/work
Initialized empty Git repository in /tmp/d20190730-915-x8jbz9/work/.git/
Switched to a new branch 'gh-pages'
cd -
cd /tmp/d20190730-915-x8jbz9/work
Aborting commit due to empty commit message.
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Couldn't push the build to github.com/.git:gh-pages
```

```console
irb(main):002:0> Shellwords.escape(nil)
=> "''"
irb(main):003:0> !!Shellwords.escape(nil)
=> true
```

Reply:
```suggestion
```